src/frontend/dbusfrontend/dbusfrontend.cpp

```diff
@@ -282,6 +282,9 @@ class DBusInputContext1 : public InputContext,
     bool processKeyEvent(uint32_t keyval, uint32_t keycode, uint32_t state,
                          bool isRelease, uint32_t time) {
         CHECK_SENDER_OR_RETURN false;
+
+        im_->instance()->setInputMethodMode(InputMethodMode::PhysicalKeyboard);
```

Comment:
If the user show the virtual keyboard by dbus, the input method mode will be OnScreenKeyboard. Then, if the user press any physical keyboard key, we want the virtual keyboard to hide automatically. That's why I should change the input method mode in this function.

Reply:
Instead, you may want to use watchEvent on KeyEvent.——Could you explain more?

Reply:
Do you mean that I should watch the event: EventType::InputContextKeyEvent, so that there is no need to duplicate code in dbusfrontend and fcitx4frontend. If so, I will move the code before into the following codes:
`d->eventWatchers_.emplace_back(
        watchEvent(EventType::InputContextKeyEvent,
                   EventWatcherPhase::InputMethod, [this](Event &event) {
                       auto &keyEvent = static_cast<KeyEvent &>(event);
                       auto *ic = keyEvent.inputContext();
                       auto *engine = inputMethodEngine(ic);
                       const auto *entry = inputMethodEntry(ic);
                       if (!engine || !entry) {
                           return;
                       }
                       engine->keyEvent(*entry, keyEvent);
                   }));`

Reply:
Yeah, and put that in a separate PR.

Reply:
> Yeah, and put that in a separate PR.

OK. It will be a very small PR.